### PR TITLE
[NIN] DT Update

### PR DESCRIPTION
**Pull request by Taurenkey**
- [X] Simple Modes updated to not break
- [X] Advanced Modes updated to not break
- [ ] Try to organize actions in a logical manner
- [X] Update opener to level 100

### CODE REVIEW

**Comment by leifktaylor** [comment]
I've been leveling Ninja from 90 to 100 using this PR.
Currently level 98.
I'll try to add some more details here as I get a clearer picture, but I have noticed that the balance opener isn't working correctly when synced in 90 dungeons (aglea)

**Comment by Taurenkey** [comment]
> I've been leveling Ninja from 90 to 100 using this PR. Currently level 98. I'll try to add some more details here as I get a clearer picture, but I have noticed that the balance opener isn't working correctly when synced in 90 dungeons (aglea).
> 
> Regardless. I'm still no. 1 dps in pretty much every aglea raid I join - so it's doing its job more or less.

That's because it's not meant for level 90 anymore as per all the description updates to it. Opener is only for level 100 currently.

**Comment by kiramon2** [comment]
How can I add/test these things before they get added to the main release/auto-update portion? Sorry, not familiar with how the repository works and can't see any guides on it on the main page....

**Comment by leiftaylorcb** [comment]
> How can I add/test these things before they get added to the main release/auto-update portion? Sorry, not familiar with how the repository works and can't see any guides on it on the main page....

The simplest way to do this would be to fork the main repo, merge Tuarenkey's ninja changes into your fork's main branch, run the build github action, and then download the zip file it creates.  Unzip it to a folder on your PC, and in the experimental tab in the dalamud settings, provide the path to the unzipped directory.  If you need more explanation I suggest hopping into the discord and asking for assistance.

**Comment by Taurenkey** [comment]
We do not provide assistance on the Discord for using non-release versions of the plugin.

File: XIVSlothCombo/Combos/JobHelpers/AST.cs [hunk 1]
```diff
@@ -52,11 +54,16 @@ private static bool BetterTargetAvailable()
                 CustomComboFunctions.OutOfRange(Balance, AST_QuickTargetCards.SelectedRandomMember))
                 return true;
 
-            Svc.Log.Debug($"Picking better?");
+            var m = AST_QuickTargetCards.SelectedRandomMember as IBattleChara;
+            if ((DrawnCard is CardType.BALANCE && CustomComboFunctions.JobIDs.Melee.Any(x => x == m.ClassJob.Id)) ||
+                (DrawnCard is CardType.SPEAR && CustomComboFunctions.JobIDs.Ranged.Any(x => x == m.ClassJob.Id)))
+                return false;
```
Comment (leifktaylor) [inline XIVSlothCombo/Combos/JobHelpers/AST.cs:60]: Are these AST changes within the scope of this NIN PR?

File: XIVSlothCombo/Combos/PvE/AST.cs [hunk 2]
```diff
@@ -30,6 +30,7 @@ internal const uint
 
             //Cards
             AstralDraw = 37017,
+            UmbralDraw = 37018,
```
Comment (leifktaylor) [inline XIVSlothCombo/Combos/PvE/AST.cs:33]: Similar to comment above, not sure how these AST changes are relevant to the focus of this NIN PR.

**Comment by Taurenkey** [comment]
@leifktaylor I'm the one ultimately responsible for merging PRs, I know there's AST stuff that has nothing to do with the PR. I don't need that sort of external review of my PRs as such, I'm fully aware of what I'm putting in.

**Comment by leifktaylor** [comment]
Looks good to me.

Tested at level 100 using the following settings:
- [x] simple mode - single target
- [x] balance opener option

Opener looks correct, and it looks as though abilities are getting used as intended.

Hit practice dummy for about 6 minutes 30 seconds.
Log output (no timestamps unfortunately):
```
You just used: Chi x1
You just used: Jin x1
You just used: Ten x1
You just used: Huton x1
You just used: Kunai's Bane x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Dokumori x1
You just used: Fleeting Raiju x1
You just used: Dream Within a Dream x1
You just used: Spinning Edge x1
You just used: Bunshin x1
You just used: Phantom Kamaitachi x1
You just used: Ten Chi Jin x1
You just used: Fuma Shuriken x1
You just used: Raiton x1
You just used: Suiton x1
You just used: Tenri Jindo x1
You just used: Fleeting Raiju x1
You just used: Meisui x1
You just used: Gust Slash x1
You just used: Zesho Meppo x1
You just used: Armor Crush x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Bhavacakra x1
You just used: Fleeting Raiju x1
You just used: Kassatsu x1
You just used: Chi x1
You just used: Jin x1
You just used: Hyosho Ranryu x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: True North x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Fleeting Raiju x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: True North x1
You just used: Armor Crush x1
You just used: Spinning Edge x1
You just used: Bhavacakra x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Ten x1
You just used: Chi x1
You just used: Jin x1
You just used: Suiton x1
You just used: Kunai's Bane x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Bhavacakra x1
You just used: Spinning Edge x1
You just used: Dream Within a Dream x1
You just used: Gust Slash x1
You just used: Armor Crush x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: True North x1
You just used: Aeolian Edge x1
You just used: Bhavacakra x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Fleeting Raiju x1
You just used: Spinning Edge x1
You just used: Kassatsu x1
You just used: Chi x1
You just used: Jin x1
You just used: Hyosho Ranryu x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Bhavacakra x1
You just used: Armor Crush x1
You just used: Bunshin x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Fleeting Raiju x1
You just used: Phantom Kamaitachi x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Ten x1
You just used: Chi x1
You just used: Jin x1
You just used: Suiton x1
You just used: Spinning Edge x1
You just used: Kunai's Bane x1
You just used: Gust Slash x1
You just used: Bhavacakra x1
You just used: Armor Crush x1
You just used: Dokumori x1
You just used: Spinning Edge x1
You just used: Zesho Meppo x1
You just used: Gust Slash x1
You just used: Bhavacakra x1
You just used: Aeolian Edge x1
You just used: Dream Within a Dream x1
You just used: Spinning Edge x1
You just used: Ten Chi Jin x1
You just used: Fuma Shuriken x1
You just used: Raiton x1
You just used: Suiton x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Tenri Jindo x1
You just used: Fleeting Raiju x1
You just used: Meisui x1
You just used: Fleeting Raiju x1
You just used: Bhavacakra x1
You just used: Gust Slash x1
You just used: Kassatsu x1
You just used: Chi x1
You just used: Jin x1
You just used: Hyosho Ranryu x1
You just used: True North x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Armor Crush x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Fleeting Raiju x1
You just used: Bhavacakra x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: True North x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Ten x1
You just used: Chi x1
You just used: Jin x1
You just used: Suiton x1
You just used: Gust Slash x1
You just used: Kunai's Bane x1
You just used: Armor Crush x1
You just used: Bhavacakra x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Bunshin x1
You just used: Phantom Kamaitachi x1
You just used: Aeolian Edge x1
You just used: Dream Within a Dream x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Fleeting Raiju x1
You just used: Aeolian Edge x1
You just used: Bhavacakra x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Kassatsu x1
You just used: Chi x1
You just used: Jin x1
You just used: Hyosho Ranryu x1
You just used: True North x1
You just used: Armor Crush x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Fleeting Raiju x1
You just used: Aeolian Edge x1
You just used: Bhavacakra x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: Armor Crush x1
You just used: Spinning Edge x1
You just used: Ten x1
You just used: Chi x1
You just used: Jin x1
You just used: Suiton x1
You just used: Kunai's Bane x1
You just used: Gust Slash x1
You just used: Bhavacakra x1
You just used: Aeolian Edge x1
You just used: Dokumori x1
You just used: Spinning Edge x1
You just used: Zesho Meppo x1
You just used: Gust Slash x1
You just used: Bhavacakra x1
You just used: Aeolian Edge x1
You just used: Spinning Edge x1
You just used: Dream Within a Dream x1
You just used: Gust Slash x1
You just used: Ten Chi Jin x1
You just used: Fuma Shuriken x1
You just used: Raiton x1
You just used: Suiton x1
You just used: Ten x1
You just used: Chi x1
You just used: Raiton x1
You just used: Tenri Jindo x1
You just used: Fleeting Raiju x1
You just used: Meisui x1
You just used: Fleeting Raiju x1
You just used: Bhavacakra x1
You just used: Armor Crush x1
You just used: Kassatsu x1
You just used: Chi x1
You just used: Jin x1
You just used: Hyosho Ranryu x1
You just used: Spinning Edge x1
You just used: Gust Slash x1
You just used: True North x1
You just used: Aeolian Edge x1
```

**Pull request merged**